frontend-public/src/components/Forms/variances/AddVarianceForm.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Field, reduxForm, change } from "redux-form";
+import { remove } from "lodash";
+import { Form, Button, Popconfirm } from "antd";
+import * as FORM from "@/constants/forms";
+import { renderConfig } from "@/components/common/config";
+import { required, maxLength } from "@/utils/Validate";
+import { resetForm } from "@/utils/helpers";
+import CustomPropTypes from "@/customPropTypes";
+import VarianceFileUpload from "@/components/Forms/variances/VarianceFileUpload";
+
+const propTypes = {
+  handleSubmit: PropTypes.func.isRequired,
+  onSubmit: PropTypes.func.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  submitting: PropTypes.bool.isRequired,
+  mineGuid: PropTypes.string.isRequired,
+  complianceCodes: CustomPropTypes.options.isRequired,
+};
+
+export class AddVarianceForm extends Component {
+  state = {
+    uploadedFiles: [],
+    documentNameGuidMap: {},
+  };
+
+  onFileLoad = (documentName, document_manager_guid) => {
+    this.state.uploadedFiles.push({ documentName, document_manager_guid });
+    this.setState(({ documentNameGuidMap }) => ({
+      documentNameGuidMap: {
+        [document_manager_guid]: documentName,
+        ...documentNameGuidMap,
+      },
+    }));
+    change("uploadedFiles", this.state.uploadedFiles);
+  };
+
+  onRemoveFile = (fileItem) => {
+    remove(this.state.uploadedFiles, { document_manager_guid: fileItem.serverId });
```

Comment:
I know there's a bug with removing files, so I pulled this up in the lodash docs, and this isn't how `remove` works: https://lodash.com/docs/#remove

It accepts a predicate function as the second argument. Since it sounded like `onRemoveFile` is never invoked anyway, that likely doesn't fix anything, but it might save you some future debugging time.

Reply:
Thank you! I was looking into the docs for remove just now